src/azure-cli/azure/cli/command_modules/storage/__init__.py

```diff
@@ -185,6 +185,13 @@ def handler(ex):
                 """
                 logger.error(message)
                 return
+            if isinstance(ex, t_error) and ex.status_code == 403 and ex.error_code == 'AuthorizationFailure':
+                message = """
+You maybe not allowed by the network rules of storage account. Please check network rule set using 'az storage account show -n accountname --query networkRuleSet'.
```

Comment:
allowed to access storage account by ... ?

Reply:
How about
> The request may be blocked by network rules of the storage account.